runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -1328,8 +1328,11 @@ static void jitMethodSampleInterrupt(J9VMThread* vmThread, IDATA handlerKey, voi
          }
 
 #if defined(J9VM_JIT_DYNAMIC_LOOP_TRANSFER)
-      if (!TR::Options::getCmdLineOptions()->getOption(TR_MimicInterpreterFrameShape) &&
-          !compInfo->getPersistentInfo()->getDisableFurtherCompilation())
+      if (!TR::Options::getCmdLineOptions()->getOption(TR_MimicInterpreterFrameShape)
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+          && !compInfo->isCheckpointInProgress()
```

Comment:
I figure it's probably ok to not acquire the comp monitor here. Even if at this point a checkpoint isn't in progress but  later it does, a compilation won't be started because of other places where the flag is checked with the monitor in hand. This is more of an optimization that statistically should be useful.

Reply:
We may need a comment here so that later-on we don't think this is a bug.

Reply:
Updated in force push https://github.com/eclipse-openj9/openj9/compare/0c28030bfa3f97874e4915151f2890621020f828..9616a2c75a1e5ac76eafde07b82f8fc7132eecf8